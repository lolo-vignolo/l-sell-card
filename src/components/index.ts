export { ProductButtons } from '../components/ProductButtons';
export { ProductImage } from '../components/ProductImage';
export { ProductTitle } from '../components/ProductTitle';
export { ProductCard } from '../components/ProductCard';
