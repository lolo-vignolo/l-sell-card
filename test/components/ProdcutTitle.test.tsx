import React from 'react';
import render from 'react-test-renderer';
import { ProductCard } from '../../src/components/ProductCard';
import { ProductTitle } from '../../src/components/ProductTitle';
import { product_1 } from '../data/product';

describe('PoductTitle', () => {
  test('show the component correctly witht presonalized title', () => {
    const wrapper = render.create(<ProductTitle name="My title" />);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('show the component correctly with default title', () => {
    const wrapper = render.create(
      <ProductCard product={product_1}>
        {() => (
          <>
            <ProductTitle />
          </>
        )}
      </ProductCard>
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
