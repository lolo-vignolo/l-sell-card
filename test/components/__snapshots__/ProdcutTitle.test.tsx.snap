// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoductTitle show the component correctly with default title 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug
  </span>
</div>
`;

exports[`PoductTitle show the component correctly witht presonalized title 1`] = `
<span
  className="productDescription undefined"
>
  My title
</span>
`;
